x-pack/plugins/monitoring/public/alerts/badge.tsx

```diff
@@ -4,187 +4,140 @@
-      }
-
-      const button = (
-        <EuiBadge
-          iconType="bell"
-          color={type}
-          onClickAriaLabel={numberOfAlertsLabel(list.length)}
-          onClick={() => setShowPopover(type)}
-        >
-          {numberOfAlertsLabel(list.length)}
-        </EuiBadge>
-      );
+  const panels = showByNode
+    ? getAlertPanelsByNode(PANEL_TITLE, alerts, stateFilter)
+    : getAlertPanelsByCategory(PANEL_TITLE, inSetupMode, alerts, alertsContext, stateFilter);
 
-      const panels = [
+  if (panels.length && !inSetupMode && panels[0].items) {
+    panels[0].items.push(
+      ...[
         {
-          id: 0,
-          title: `Alerts`,
-          items: list.map(({ alert, alertState }, index) => {
-            return {
-              name: (
-                <Fragment>
-                  <EuiText size="s">
-                    <h4>{getDateFromState(alertState)}</h4>
-                  </EuiText>
-                  <EuiText>{alert.rawAlert.name}</EuiText>
-                </Fragment>
-              ),
-              panel: index + 1,
-            };
-          }),
+          isSeparator: true as const,
         },
-        ...list.map((alertStatus, index) => {
-          return {
-            id: index + 1,
-            title: getDateFromState(alertStatus.alertState),
-            width: 400,
-            content: <AlertPanel alert={alertStatus.alert} alertState={alertStatus.alertState} />,
-          };
-        }),
-      ];
-
-      badges.push(
-        <EuiPopover
-          id="monitoringAlertMenu"
-          button={button}
-          isOpen={showPopover === type}
-          closePopover={() => setShowPopover(null)}
-          panelPaddingSize="none"
-          anchorPosition="downLeft"
-        >
-          <EuiContextMenu initialPanelId={0} panels={panels} />
-        </EuiPopover>
-      );
-    }
+        {
+          name: (
+            <EuiSwitch
+              checked={false}
+              onChange={() => setShowByNode(!showByNode)}
+              label={showByNode ? GROUP_BY_TYPE : groupByType}
+            />
+          ),
+        },
+      ]
+    );
   }
 
+  const button = (
+    <EuiBadge
+      iconType="bell"
+      color={inSetupMode ? 'default' : 'danger'}
+      onClickAriaLabel={numberOfAlertsLabel(alertCount)}
+      onClick={() => setShowPopover(true)}
+    >
+      {numberOfAlertsLabel(alertCount)}
+    </EuiBadge>
+  );
+
   return (
-    <EuiFlexGrid data-test-subj="monitoringSetupModeAlertBadges">
-      {badges.map((badge, index) => (
-        <EuiFlexItem key={index} grow={false}>
-          {badge}
-        </EuiFlexItem>
-      ))}
-    </EuiFlexGrid>
+    <EuiPopover
+      id="monitoringAlertMenu"
+      button={button}
+      isOpen={showPopover === true}
+      closePopover={() => setShowPopover(null)}
+      panelPaddingSize="none"
+      anchorPosition="downLeft"
+    >
+      <EuiContextMenu
+        key={`${showByNode ? 'byNode' : 'byType'}_${panels.length}`}
```

Comment:
We don't need a `key` here since since it's not iterated (at its nested scope).

Reply:
Sure, I can try to remove it. I originally added it because I saw strange behavior when toggling between the modes and React refusing to render any changes.

Reply:
I ended up adding this back in because I saw the same behavior. I'd be happy to solve the issue another way if you have ideas.